handlers/create.go

```diff
@@ -59,12 +65,13 @@ func CreateMachine(event *events.Event, apiClient *client.RancherClient) error {
 		return err
 	}
 
-	errString := ""
-	go logProgress(readerStdout, readerStderr, publishChan, machine, event, &errString)
+	errChan := make(chan string, 1)
+	go logProgress(readerStdout, readerStderr, publishChan, machine, event, errChan)
 
 	err = command.Wait()
 
 	if err != nil {
+		errString := <-errChan
```

Comment:
This could potentially lead to deadlock. I would put the channel read in a `select` with a short timeout.